File: Python/Code Snippets/LINEAR SEARCH.py
#TO SEARCH AN ELEMENT FROM A LIST
#AUTHOR: ASWIN P
#DATE: 12/7/2022

#FUNCTION
def search(l,s):
    it=0
    pos=[]
    for i in range(len(l)):
        if l[i]==s:
            it+=1
            pos.append(i+1)
    if it==0:
        print("element",s,"is NOT present")
    else:
        print("element",s,"is PRESENT",it,"time(s) at the position(s)",pos)

#PROGRAM
n=int(input("ENTER THE NO.OF ELEMENTS: "))
list=[]
for i in range(n):
    s=int(input("ENTER THE ELEMENT: "))
    list.append(s)
item=int(input("ENTER THE ITEM TO SEARCH: "))
search(list,item)

File: C/Code Snippetts/linear_search.c
//program to implement linear search
#include<stdio.h>

void read_array(int a[], int n){
    printf("ENTER ELEMENTS OF THE ARRAY\n");
    for(int i = 0; i < n; i++){
        printf("ELEMENT %d: ", i+1);
        scanf("%d", &a[i]);
    }
}


int linear_search(int a[], int key, int n){
    int found = 0;
    for(int i = 0; i < n; i++){
        if (a[i] == key){
            found = i + 1;
            break;
        }
    }
    return found;
}


int main(){
    int arr[20], n, key, found;
    printf("ENTER NO. OF ELEMENTS: ");
    scanf("%d",&n);
    read_array(arr, n);
    printf("ENTER KEY TO SEARCH IN ARRAY: ");
    scanf("%d",&key);
    found = linear_search(arr, key, n);
    if (found)
        printf("%d FOUND AT POSITION %d", key, found);
    else
        printf("%d NOT FOUND", key);
    return 0;
}